Make this a pip-installable package (#10)
fmauch/catmux: README.md

one central spot on a particular machine.

## Installation
Catmux is a pure python package and as such it is installable via pip.
After cloning this repository call `pip install --user .` in the repository's root directory.

## Usage
Currently, there is no full-blown documentation, but the example config file in
`etc/example_session.yaml` gives a detailed insight on possible commands.

### Running the example the most simple way

> After installation, you can run a simple example by calling the following command in the
> repository's root directory.

Reviewer: ```suggestion
After installation, you can run a simple example by calling one of the following commands.
```

                        using a comma-separated list such as '--overwrite param_a=1,param_b=2'.
```

### Full blown example
To make use of all catmux features, run the following example command, again from the root
directory:
```
catmux_create_session etc/example_session.yaml --tmux_config etc/tmux_default.conf --session_name example_session --overwrite show_layouts=True,replacement_param="new catmux user"
```

### Killing a catmux session
If you are not that familiar with tmux: To kill a session, simply type `tmux kill-session` in any
terminal window. In the `etc/tmux_default.conf` there is a key-binding for that, see
`etc/readme_tmux.txt` for details.

## Migrating from the catkin version of catmux
With the spread of ROS2, the need for a catkin-independent catmux has emerged.
Catmux is now a plain python package without the ROS integration.
Therefore, it is installed via pip and no longer needs to be part of a catkin workspace.
This opens the possibility to easily use catmux outside of robotics application, e.g. for server
administration or other development tasks.

Catmux no longer has the `package://<your_package_name>` lookup capabilities.

> To archive the same outcome, simply use rospack:

Author: ```suggestion
To achieve the same outcome, simply use rospack:
```

                        This config will be used for the tmux session
  -d, --detach          Start session in detached mode
  --overwrite OVERWRITE
                        Overwrite a parameter from the session config. Parameters can be specified
                        using a comma-separated list such as '--overwrite param_a=1,param_b=2'.
```

### Full blown example

> To make use of all catmux features, run the following example command, again from the root
> directory:

Reviewer: ```suggestion
To make use of all catmux features, run one of the following example commands:
```

```

> catmux_create_session ${HOME}/.local/share/catmux/example_session.yaml \
>   --tmux_config {HOME}/.local/share/catmux/tmux_default.conf \
>   --session_name example_session \
>   --overwrite show_layouts=True,replacement_param="new catmux user"

Reviewer: ```suggestion
# In case you used the `--user` flag during the install process
catmux_create_session ${HOME}/.local/share/catmux/example_session.yaml \
  --tmux_config {HOME}/.local/share/catmux/tmux_default.conf \
  --session_name example_session \
  --overwrite show_layouts=True,replacement_param="new catmux user"
  
  
# In case you used `sudo pip install ...` for installation
catmux_create_session $(python3 -c "import sys; print(sys.prefix)")/share/catmux/example_session.yaml \
  --tmux_config $(python3 -c "import sys; print(sys.prefix)")/share/catmux/tmux_default.conf \
  --session_name example_session \
  --overwrite show_layouts=True,replacement_param="new catmux user"
```

Reviewer: Using `--user` the full-blown example from the README works.
